static/app/views/insights/common/queries/useSpansQuery.tsx

```diff
@@ -50,58 +40,104 @@ export function useSpansQuery<T = any[]>({
   queryExtras?: SpansRPCQueryExtras;
   referrer?: string;
   trackResponseAnalytics?: boolean;
-}) {
-  const isTimeseriesQuery = (eventView?.yAxis?.length ?? 0) > 0;
-  const queryFunction = isTimeseriesQuery
-    ? useWrappedDiscoverTimeseriesQuery
-    : useWrappedDiscoverQuery;
+};
 
+export function useSpansQuery<T = any[]>({
+  referrer = 'use-spans-query',
+  trackResponseAnalytics = true,
+  ...props
+}: SpansQueryProps<T>) {
   const {isReady: pageFiltersReady} = usePageFilters();
+  return useSpansQueryBase({
+    ...props,
+    enabled: (props.enabled || props.enabled === undefined) && pageFiltersReady,
+    referrer,
+    trackResponseAnalytics,
+    withPageFilters: true,
```

Comment:
nit: Instead of building out separate functions for with and without page filters, could we just add `withPageFilters` as a prop to `SpansQueryProps` (with a default value of `true`)? That way, any time `useSpansQuery` is called, the user can set `withPageFilters` to true or false, and the setting can propagate to `useWrappedDiscoverTimeseriesQuery` or `useWrappedDiscoverQuery`.

Reply:
Yep probably a bit of a stretch to add separate functions here I'll follow up with a PR